perllib/FixMyStreet/App/Controller/Report/New.pm

```diff
@@ -1014,8 +1014,15 @@ sub process_user : Private {
         my $user = $c->user->obj;
         $report->user( $user );
         $report->name( $user->name );
-        $c->stash->{check_name} = 1;
-        $c->stash->{login_success} = 1;
+
+        # iOS app can't send cookies, so instead sends
+        # user's email/password when making a new report.
+        # In that case we can't check the user's name so
```

Comment:
I see what this is for, but I would like to check the name/anonymous if possible I think. How about if the mobile app sends a '2' if logged in, but remains a 1 if submit_sign_in clicked, and then when it's 2 it doesn't do check_name/login_success then. Does that work okay?